x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java

```diff
@@ -498,34 +496,19 @@ public void prefetchPart(final int part) throws IOException {
                     // We do not actually read anything, but we want to wait for the write to complete before proceeding.
                     // noinspection UnnecessaryLocalVariable
                     final Tuple<Long, Long> rangeToRead = rangeToWrite;
-
-                    try (Releasable ignored = cacheFile.fileLock()) {
```

Comment:
This lock seems pointless to me and was causing trouble. If an eviction of the current file is concurrently started, all that happened here was that we would block that eviction when as far as I can tell we should simply fail the cache population as it's just wasted effort anyway as soon as the eviction hits?